### ISSUE TITLE
Type '"batch"' is not assignable to type '"fine-tune" | "assistants"

### ISSUE THREAD

**Issue by euclid1990** [openai/openai-node#830]
### Confirm this is a Node library issue and not an underlying OpenAI API issue

- [X] This is an issue with the Node library

### Describe the bug

According to the document [Uploading Your Batch Input File](https://platform.openai.com/docs/guides/batch/2-uploading-your-batch-input-file).  I tried the following code:

```js
import fs from "fs";
import OpenAI from "openai";

const openai = new OpenAI();

async function main() {
  const file = await openai.files.create({
    file: fs.createReadStream("batchinput.jsonl"),
    purpose: "batch",
  });

  console.log(file);
}

main();
```

However, it outputs the error:

```
TSError: ⨯ Unable to compile TypeScript:
worker/generate.ts:18:4 - error TS2322: Type '"batch"' is not assignable to type '"fine-tune" | "assistants"'.

18    purpose: "batch",
      ~~~~~~~

  node_modules/openai/resources/files.d.ts:119:5
    119     purpose: 'fine-tune' | 'assistants';
            ~~~~~~~
    The expected type comes from property 'purpose' which is declared here on type 'FileCreateParams'
```

Maybe the type *batch* is missing in this line of the openai package source code:

https://github.com/openai/openai-node/blob/7196ac9310d58d057fb2a575e60c1718bf6341a2/src/resources/files.ts#L185

We can fix it by adding `batch`

```js
purpose: 'fine-tune' | 'assistants' | 'batch';
```  

### To Reproduce

Easy to reproduce: look at official documentation page - https://platform.openai.com/docs/guides/batch/2-uploading-your-batch-input-file
Run above code in nodejs / typescript.

### OS

macOS

### Node version

v20.11.0

### Library version

v4.42.0

**Comment by pstern-sl** [comment]
Thanks for flagging. We will take a look!

**Comment by pstern-sl** [comment]
This is resolved in 4.43 which should be merged shortly! Thank you for reporting. 
https://github.com/openai/openai-node/pull/832
